spec/unit/node_spec.rb

```diff
@@ -445,25 +447,30 @@ def from_json(json)
 
   describe "and :node_name is set to 'cert'" do
     before do
-      Puppet[:strict_hostname_checking] = false
       Puppet[:node_name] = "cert"
+      @node = Puppet::Node.new("foo.domain.com",
+                               :parameters => {"hostname" => "yay", "domain" => "domain.com"})
     end
 
     it "uses the passed-in key as the first value" do
       expect(@node.names[0]).to eq("foo.domain.com")
     end
 
     describe "and strict hostname checking is enabled" do
-      it "only uses the passed-in key" do
+      before do
         Puppet[:strict_hostname_checking] = true
+        @node = Puppet::Node.new("foo.domain.com",
```

Comment:
same here

Reply:
Good catch. I re-wrote the setup and teardown of these specs in an intermediate step and forgot they weren't needed when I cleaned them up.

Reply:
pushed an update addressing your comments